File: jpa-java-example/src/main/java/com/example/jpajavaexample/domain/user/repository/UserRepository.java
package com.example.jpajavaexample.domain.user.repository;

import com.example.jpajavaexample.domain.user.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserRepository {

    boolean existsByEmail(String email);

    User save(User user);

    Page<User> findAll(Pageable pageable);
}

File: graphql-example/src/main/kotlin/com/example/graphqlexample/repository/UserRepository.kt
package com.example.graphqlexample.repository

import com.example.graphqlexample.domain.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun findByNameContainingIgnoreCase(keyword: String): List<User>
    fun existsByEmail(email: String): Boolean
}
